cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsCredentialConnector.java

```diff
@@ -125,19 +125,24 @@ public CloudCredentialStatus delete(AuthenticatedContext auth) {
     }
 
     @Override
-    public CredentialPrerequisitesResponse getPrerequisites(CloudContext cloudContext, String externalId, String deploymentAddress, CredentialType type) {
+    public CredentialPrerequisitesResponse getPrerequisites(CloudContext cloudContext, String externalId,
+        String auditExternalId, String deploymentAddress, CredentialType type) {
         String policyJson;
+        String actualExternalId;
         switch (type) {
             case ENVIRONMENT:
                 policyJson = awsPlatformParameters.getCredentialPoliciesJson();
+                actualExternalId = externalId;
                 break;
             case AUDIT:
                 policyJson = awsPlatformParameters.getAuditPoliciesJson();
+                actualExternalId = auditExternalId;
                 break;
             default:
                 policyJson = null;
+                actualExternalId = null;
         }
-        AwsCredentialPrerequisites awsPrerequisites = new AwsCredentialPrerequisites(externalId, policyJson);
+        AwsCredentialPrerequisites awsPrerequisites = new AwsCredentialPrerequisites(actualExternalId, policyJson);
```

Comment:
it can be null?

Reply:
it can if we introduce a new credential type